### PR TITLE
[MRG+1] Add notes for Python 3 transition

### LINKED ISSUE #5167: Python 2 support roadmap
What's the roadmap towards sunsetting Python 2 support for MNE? Many projects have already decided to support only Python 3, with one final Python 2 release that won't get any new features. MNE should be on the http://python3statement.org/ website, which means that we will have to drop Python 2 support by 2020 at the latest. Many projects will do or have done this earlier by releasing one final Python 2-compatible version, while development continues with Python 3-only code.

### PR DESCRIPTION
**Pull request by cbrnr**
Fixes #5167.

### CODE REVIEW

**Comment by cbrnr** [comment]
Maybe highlight this in `README.rst` and https://8201-1301584-gh.circle-artifacts.com/0/html/install_mne_python.html? If so, how?

File: README.rst [hunk 1]
```diff
@@ -61,6 +61,8 @@ To install the latest stable version of MNE-Python, you can use `pip <https://pi
 
     pip install -U mne
 
+Note that MNE-Python 0.17 will be the last release to support Python 3. From MNE-Python 0.18, only Python 3 will be supported.
```
[the remark this answers is not in the document]
Comment (cbrnr) [reply README.rst]: Yep 😉

File: README.rst [hunk 2]
```diff
@@ -61,6 +61,8 @@ To install the latest stable version of MNE-Python, you can use `pip <https://pi
 
     pip install -U mne
 
+Note that MNE-Python 0.17 will be the last release to support Python 2. From MNE-Python 0.18, only Python 3 will be supported.
```
Comment (larsoner) [inline README.rst]: I'd do `**Note**` to make it stand out a bit

File: doc/install_mne_python.rst [hunk 3]
```diff
@@ -5,87 +5,77 @@
...
-      $ conda --version && python --version
-      conda 4.4.10
-      Python 3.6.4 :: Continuum Analytics, Inc.
+If you get an error message, consult the Anaconda documentation and search for Anaconda install
+tips (StackExchange results are often helpful).
 
-  If it doesn't, something went wrong.
-  Look through the Anaconda documentation and Google Anaconda install
-  tips (StackExchange results are often helpful).
+.. note:: Note that MNE-Python 0.17 will be the last release to support Python 2. From MNE-Python 0.18, only Python 3 will be supported.
```
Comment (larsoner) [inline doc/install_mne_python.rst]: this `.. note::` directive should be sufficient to highlight it

**Comment by cbrnr** [comment]
Looks good: 
https://8206-1301584-gh.circle-artifacts.com/0/html/install_mne_python.html
https://github.com/cbrnr/mne-python/blob/1e754b257780a7d25b2b9de0c2693fece8d2a63e/README.rst

File: doc/install_mne_python.rst [hunk 4]
```diff
@@ -5,87 +5,77 @@
...
+
+    $ conda --version && python --version
+    conda 4.5.4
+    Python 3.6.5 :: Anaconda, Inc.
 
-      $ conda --version && python --version
-      conda 4.4.10
-      Python 3.6.4 :: Continuum Analytics, Inc.
+If you get an error message, consult the Anaconda documentation and search for Anaconda install
+tips (StackExchange results are often helpful).
```
Comment (christianbrodbeck) [inline doc/install_mne_python.rst]: What about turning `StackExchange` into a link?

**Comment by larsoner** [comment]
I prefer the old bullet point format, it makes it clear which phrases all go together (e.g., macOS users only section, Windows users only section)

**Comment by cbrnr** [comment]
I didn't like the bullet points because the text reads easier without them. What about putting the OS-specific remarks in a note box?

**Comment by larsoner** [comment]
IIRC I tried it (and it's a pain because of the font-awesome you have to do the HTML yourself) and it was too distracting

**Comment by larsoner** [comment]
... but feel free to try it if you want

**Comment by cbrnr** [comment]
Yes, the manual HTML is not ideal. If you really want the icons, what about using https://github.com/fraoustin/sphinx_fontawesome? Or if we can live without the icons (IMO they are nice but not really worth introducing ugly HTML) we could just not have them.

**Comment by larsoner** [comment]
I had no idea sphinx-fontawesome existed, yes let's use that (and please replace any other existing occurrences if you have time).

**Comment by cbrnr** [comment]
Do I have to add this module to the `pip` section of `environment.yml` (and `environment2.yml`)?

**Comment by larsoner** [comment]
We could vendor it in `doc/sphinxext` but I think `pip` is better, yeah

**Comment by cbrnr** [comment]
I think this would look really nice if the color scheme in the code block actually made the text visible:
https://8211-1301584-gh.circle-artifacts.com/0/html/install_mne_python.html

Any ideas?

**Comment by larsoner** [comment]
In `_static/reset_style.css` set `color: #2c3e50` instead of `inherit`, and add a comment that it's explicitly setting it to the color of the Pygments option we use

**Comment by agramfort** [comment]
there is a a rendering bug in blue boxes:

https://8211-1301584-gh.circle-artifacts.com/0/html/install_mne_python.html

**Comment by larsoner** [comment]
Fixed:

https://8217-1301584-gh.circle-artifacts.com/0/html/install_mne_python.html

**Comment by cbrnr** [comment]
Thanks @larsoner - looks very good! I guess the two Travis jobs have died for another reason (timeout).

File: doc/install_mne_python.rst [hunk 5]
```diff
@@ -5,87 +5,72 @@
...
-      $ conda --version && python --version
-      conda 4.4.10
-      Python 3.6.4 :: Continuum Analytics, Inc.
+If you get an error message, consult the Anaconda documentation and search for Anaconda install
+tips (`Stack Overflow <https://stackoverflow.com/>`_ results are often helpful).
 
-  If it doesn't, something went wrong.
-  Look through the Anaconda documentation and Google Anaconda install
-  tips (StackExchange results are often helpful).
+.. note:: Note that MNE-Python 0.17 will be the last release to support Python 2. From MNE-Python 0.18, only Python 3 will be supported.
```
Comment (larsoner) [inline doc/install_mne_python.rst]: You do not need "Note that", it's implied by the "Note" heading that is automatically added by the `.. note::` directive

**Comment by cbrnr** [comment]
I'm not a huge fan of these bullet points, they don't seem necessary when there's mostly only 1 or 2 items in the list (and that's why I removed them of course). However, I agree that the admonition boxes are too prominent.

I'll try subsections next. Do you have other ideas how to keep a nice structure without bullet points?

**Comment by cbrnr** [comment]
There are also several more preset admonitions (http://docutils.sourceforge.net/docs/ref/rst/directives.html#admonitions), maybe one of them fits and renders a bit more subtle?

**Comment by cbrnr** [comment]
I think it should be possible to render admonitions with a custom style: https://stackoverflow.com/a/9738992/1112283

Copying the `note` style would probably work, just need to figure out how.

**Comment by cbrnr** [comment]
Nice, it works: https://8221-1301584-gh.circle-artifacts.com/0/html/install_mne_python.html

WDYT?

**Comment by larsoner** [comment]
Yes that's better, LGTM +1 for merge

**Comment by cbrnr** [comment]
Is the CSS change still necessary or should we revert it?

**Comment by larsoner** [comment]
You can answer this question locally by viewing the page in chrome, selecting the text of interest, right click, inspect, alter the CSS that you want (here, set the highlight color in reset-style.css to `inherit`)

**Comment by cbrnr** [comment]
Thanks @larsoner - all good now, ready to merge! https://8229-1301584-gh.circle-artifacts.com/0/html/install_mne_python.html

**Comment by cbrnr** [comment]
Maybe one last question regarding the dependencies in `README.rst`. Currently, we list Python >= 2.7. Should we add a clarifying note (in parens) that this will change in MNE 0.18? Also, what will be the required Python 3 version for MNE 0.18? I'd suggest the latest Python version (which will be at least 3.7 by the time 0.18 is released).

**Comment by agramfort** [comment]
please let's be too bleeding edge. You have to accept that users have admin
constraints and cannot install latest versions of everything. I've seen
this in many places.

I would suggest >= 3.4 like sklearn does.

**Comment by cbrnr** [comment]
@agramfort sticking with what scikit-learn does sounds good to me! We still have some time until MNE 0.18 will be released, so by that time scikit-learn might also require a later Python version. We don't have to mention the exact Python 3 version right now anyway.

**Comment by cbrnr** [comment]
(But the admin argument does not hold if users install Anaconda as we recommend - Anaconda does not require admin privileges by default!)

**Comment by agramfort** [comment]
see that I fought to make sure mne is 80% usable with just numpy and scipy
:)
not everyone uses anaconda...

**Comment by cbrnr** [comment]
😄 you're right, I don't use Anaconda for example (but Homebrew doesn't need admin privileges either 😛).

Regarding NumPy and SciPy as the only hard dependencies, is #5248 now fixed (i.e. does `import mne` produce no error if the user does not have `matplotlib` installed)?

**Comment by agramfort** [comment]
yes matplotlib is not anymore a hard dep

**Comment by cbrnr** [comment]
Alright then, let's wait for @larsoner if he's happy, then either one of you feel free to merge. Afterwards, I'll add MNE to the python3statement website.

**Comment by larsoner** [comment]
Thanks @cbrnr 

Yes `mpl` should not be required anymore

**Pull request merged**